src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -321,6 +338,16 @@ private ConnectionState doReadHTTPInitial(HttpRequest httpRequest) {
         }
     }
 
+    private boolean isProxyRequested() {
+        // HTTPS requests have uri without http scheme too
+        if (currentRequest.getMethod() == HttpMethod.CONNECT || sslEngine != null) {
+            return false;
+        }
+        // direct requests to the proxy have the path only without a scheme
```

Comment:
Yes, I've tried to request the proxy with its external address. The URI 
has no scheme, host and port too.

Am 19.07.2015 um 21:40 schrieb Jason Hoetger:
> In src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java
> <https://github.com/adamfisk/LittleProxy/pull/220#discussion_r34961261>:
>
>> @@ -321,6 +338,16 @@ private ConnectionState doReadHTTPInitial(HttpRequest httpRequest) {
>>          }
>>      }
>>
>> +    private boolean isProxyRequested() {
>> +        // HTTPS requests have uri without http scheme too
>> +        if (currentRequest.getMethod() == HttpMethod.CONNECT || sslEngine != null) {
>> +            return false;
>> +        }
>> +        // direct requests to the proxy have the path only without a scheme
>
> Is this statement accurate? If the proxy were running at (for example)
> 192.168.100.100, and I made an HTTP request to http://192.168.100.100,
> wouldn't request.getUri() return "http://192.168.100.100"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adamfisk/LittleProxy/pull/220/files#r34961261>.
>